janusgraph-solr/src/test/java/org/janusgraph/diskstorage/solr/SolrIndexTest.java

```diff
@@ -82,6 +89,9 @@ protected Configuration getLocalSolrTestConfig() {
 
         config.set(SolrIndex.ZOOKEEPER_URL, SolrRunner.getZookeeperUrls(), index);
         config.set(SolrIndex.WAIT_SEARCHER, true, index);
+        if(testingStaleValues) {
```

Comment:
Didn't yet check the test but it looks confusing. Method `@BeforeAll` is executed only once before all tests. So, it always will be false here.
Instead, you should probably use parametrized test.

Reply:
I'm confused by what you mean here- `testingStaleValues` is initialized as false outside of the `@BeforeAll ` method. It is set as true within the one function (`testUpdateListWithNoStaleValues`) that uses it, but is otherwise false because I didn't want to affect other tests with the altered config. 
Within `testUpdateListWithNoStaleValues`, SolrIndex is created with the remove-stale-values flag marked as true when `index = new SolrIndex(getLocalSolrTestConfig());` and `clopen()` are called.

Reply:
Sorry, I didn't check the code properly. Now I get what you want to do here.